### PR TITLE
DashR Shiny Diamonds Explorer App

### PR DESCRIPTION
**Pull request by milaroisin**
Issue for app: https://github.com/plotly/dash-sample-apps/issues/177

# App pull request
- [ ] This is a new app
- [x] I am improving an existing app (redesigns/code "makeovers")

## About

- Playground deployment URL (new version):
- Current gallery app URL: (delete this line if inapplicable)
- Python app repository link: (delete this line if you are working on a Python app)

## Workflow

- [x] I have created a branch in the appropriate monorepo, and the
  elements necessary for successful deployment are in place.
- [x] If the app is a redesigned and/or restyled version of an
  existing gallery app, I've summarized the changes requested in the
  appropriate Streambed issue and confirm that they have been applied.
- [x] If the app is on the Dash Gallery portal, I have added a link to
  the GitHub repository for the source code in the portal description.
- [x] If the app is a reimplementation of a Python gallery app for the
  DashR gallery, the app in this PR mimics, as closely as possible,
  the style and functionality of the existing app.=
- [x] I have removed *all Google Analytics code* from the app's
  `assets/` folder.

## The pre-review review

I have addressed all of the following questions:

- [x] Does everything in my code serve some purpose? (I have removed
  any dead and/or irrelevant code.)
- [x] Does everything in my code have a clear purpose? (My code is
  readable and, where it isn't, it has been commented appropriately.)]
- [x] Am I reinventing the wheel? (I have used appropriate packages to
  lessen the volume of code that needs to be maintained.)

### CODE REVIEW

**Comment by milaroisin** [comment]
@rpkyle We figured out the problem, but as mentioned before in Slack, if we chose to keep the numeric labels of the app in the French comma format to replicate the Shiny version in DashR, the labels get squished together. Other than that, the app is completed.

**Comment by CanerIrfanoglu** [comment]
Committed 2091fbf6ba8eed6b1ee6c9a8503560e659bb51ac. 

The app was already functional before this commit. Just made small performance and code readability improvement changes 

- Used `lapply` for `DropDownMenuOptions`
- Broke app sections with proper commenting
- Removed redundant code from `app$layout`
- Linted the app

File: apps/dashr-shiny-diamond-explorer/app.R [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    #FacetGrid ggplot2 object
+    p <- ggplot(dataset, aes_string(x = x, y = y, color = color)) + geom_point()
+
+    # Add it if least one facet column/row is specified
+    facets <- paste(facet_row, '~', facet_col)
+
+
+    if (facets != '. ~ .') { p <- p + facet_grid(facets)
+      p1 <- ggplotly(p) %>%
+      layout(height = plotHeight,
```
Comment (HammadTheOne) [inline apps/dashr-shiny-diamond-explorer/app.R]: Fixed `height` argument to specify in the `ggplotly` function, specifying it in `layout()` is deprecated.

File: apps/dashr-shiny-diamond-explorer/app.R [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+               input(id = 'facet-column-dropdown', property = 'value')
+             ),
+             
+  function(sampleSize, plotHeight, x, y, color, facet_row,facet_col) {
+
+    #Adding Reactive data info, dataset is built in diamonds data
+    dataset <- diamonds[sample(nrow(diamonds), sampleSize),]
+
+    #FacetGrid ggplot2 object
+    p <- ggplot(dataset, aes_string(x = x, y = y, color = color)) + geom_point()
```
Comment (HammadTheOne) [inline apps/dashr-shiny-diamond-explorer/app.R]: Renamed the plot variables to be a little bit more descriptive.

**Comment by josegonzalez** [comment]
As per https://github.com/plotly/streambed/issues/13854, I am closing this as the new deployment process is in place.

If you need any changes from this branch, please recreate your changes against master and create a new pull request. Let me know if you need help and I am happy to jump in.

**Pull request closed without merging**